kubernetes/windows/hostlogs/main.ps1

```diff
@@ -0,0 +1,105 @@
+    if (Test-Path $schemaVersionFile) {
+        $schemaVersion = Get-Content $schemaVersionFile | ForEach-Object { $_.TrimEnd() }
+        if ($schemaVersion.GetType().Name -eq 'String') {
+            [System.Environment]::SetEnvironmentVariable("AZMON_AGENT_CFG_SCHEMA_VERSION", $schemaVersion, "Process")
+            [System.Environment]::SetEnvironmentVariable("AZMON_AGENT_CFG_SCHEMA_VERSION", $schemaVersion, "Machine")
+        }
+        $env:AZMON_AGENT_CFG_SCHEMA_VERSION
+    }
+
+    # Set env vars for geneva monitor
+    $envVars = @{
+        MONITORING_DATA_DIRECTORY = (Join-Path $rootDir "opt\genevamonitoringagent\datadirectory")
+        MONITORING_GCS_AUTH_ID_TYPE = "AuthMSIToken"
+        MONITORING_GCS_REGION = $aksregion
+        MONITORING_TENANT = "CloudAgent"
+        MONITORING_ROLE = "Windows-HPC-Geneva"
+        MONITORING_ROLE_INSTANCE = "$hostName-$podName"       
+    }
+
+    foreach($key in $envVars.PSBase.Keys) {
+        [System.Environment]::SetEnvironmentVariable($key, $envVars[$key], "Process")
+        [System.Environment]::SetEnvironmentVariable($key, $envVars[$key], "Machine")
+    }
+
+    # run config parser
+    $rubypath =  "./ruby31/bin/ruby.exe"
+
+    #Parse the configmap to set the right environment variables for geneva config.
+    & $rubypath ./opt/hostlogswindows/scripts/ruby/tomlparser-hostlogs-geneva-config.rb
+    .\setagentenv.ps1
+}
+
+function Get-GenevaEnabled {
+  $gcsEnvironment = [System.Environment]::GetEnvironmentVariable("MONITORING_GCS_ENVIRONMENT", "process")
+  $gcsAccount = [System.Environment]::GetEnvironmentVariable("MONITORING_GCS_ACCOUNT", "process")
+  $gcsNamespace = [System.Environment]::GetEnvironmentVariable("MONITORING_GCS_NAMESPACE", "process")
+  $gcsConfigVersion = [System.Environment]::GetEnvironmentVariable("MONITORING_CONFIG_VERSION", "process")
+  $gcsAuthIdIdentifier = [System.Environment]::GetEnvironmentVariable("MONITORING_MANAGED_ID_IDENTIFIER", "process")
+  $gcsAuthIdValue = [System.Environment]::GetEnvironmentVariable("MONITORING_MANAGED_ID_VALUE", "process")
+
+  return (![string]::IsNullOrEmpty($gcsEnvironment)) -and 
+    (![string]::IsNullOrEmpty($gcsAccount)) -and 
+    (![string]::IsNullOrEmpty($gcsNamespace)) -and 
+    (![string]::IsNullOrEmpty($gcsConfigVersion)) -and 
+    (![string]::IsNullOrEmpty($gcsAuthIdIdentifier))  -and 
+    (![string]::IsNullOrEmpty($gcsAuthIdValue)) 
+}
+
+Start-Transcript -Path main.txt
+
+Set-EnvironmentVariables
+Start-FileSystemWatcher
+
+if(Get-GenevaEnabled){
+    Invoke-Expression ".\opt\genevamonitoringagent\genevamonitoringagent\Monitoring\Agent\MonAgentLauncher.exe -useenv"
+} else {
+    Write-Host "Geneva not configured. Watching for config map"
+    # Infinite loop keeps container alive while waiting for config map
+    # Otherwise when the process ends, kubernetes sees this as a crash and the container will enter a crash loop
+    while($true){
+        Start-Sleep 5
```

Comment:
We should use ping or better way to prevent windows container stopping. Lets ping windows team to find out better guidance here and will share the windows engineering team contact offline.

Reply:
Sounds good.

Reply:
We could move the FileSystemWatcher to the main process in this case. If we're not starting the agent, then there's no need to create a separate process for the FileSystemWatcher. And by running it in the main process, it'll keep the process alive

Reply:
Increased sleep duration to 3600 (1 hour). I'm seeing 0 cpu utilization during the sleep